### PR TITLE
chore: initial-setup-for-davinci-client

### PR DESCRIPTION
**Pull request by ryanbas21**
setup the davinci client in the sdk repo. Linting / testing configuration. No real changes beyond that

There are lint errors in this branch.

# JIRA Ticket

Please link jira ticket here

# Description

# Type of Change

Please Delete options that are not relevant

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

# Definition of Done

Check all that apply

- [ ] Acceptance criteria is met.
- [ ] All tasks listed in the user story have been completed.
- [ ] Coded to standards.
- [ ] Code peer-reviewed.
- [ ] Ensure backward compatibility (special attention).
- [ ] API reference docs is updated.
- [ ] Unit tests are written.
- [ ] Integration tests are written.
- [ ] e2e tests are written.
- [ ] CI build passing on the feature branch.
- [ ] Functional spec is written/updated
- [ ] contains example code snippets.
- [ ] Change log updated.
- [ ] Documentation story is created and tracked.
- [ ] UI is completed or ticket is created.
- [ ] Demo to PO and team.
- [ ] Tech debts and remaining tasks are tracked in separated ticket(s).

## Documentation

- [ ] Acceptance criteria met
- [ ] Spell-check run
- [ ] Peer reviewed
- [ ] Proofread

### CODE REVIEW

File: nx.json [hunk 1]
```diff
@@ -73,9 +73,10 @@
   },
   "generators": {},
   "release": {
-    "projects": ["packages/*"],
+    "projects": ["packages/*", "davinci-client-davinci-client"],
```
Comment (ryanbas21) [inline nx.json:76]: I don't think we need this here

**Review by ryanbas21 — COMMENTED** [review]
One thought I had while reading this, is if its worth looking into using Branded types, at least for internal api functions.

Branded types would allow us to differ between two types that are the same. For example 

we have things like 'id' and 'type' that can be a string. We can brand these types with a symbol so that even ifthe value is a string, it has to be branded as an ID.

It adds a layer of type safety. Not something I think needs to be in this PR, but something to think about. I don't think we should necessarily do it at the API level because then we would have to export brands for people to import and use I think.

File: packages/davinci-client/README.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+import { davinciClient } from '@forgerock/davinci';
+
+const davinci = await davinciClient({
+  config: (getDefaultConfig) => {
+    return getDefaultConfig({
+      baseUrl: 'https://auth.pingone.ca/02fb4743-189a-4bc7-9d6c-a919edfe6447/',
+      // baseUrl: 'https://auth.test-one-pingone.com/0c6851ed-0f12-4c9a-a174-9b1bf8b438ae/',
+      clientId: '724ec718-c41c-4d51-98b0-84a583f450f9',
+      // clientId: '8d3948bc-de0c-4b85-88c4-35b750f2ce47',
+      redirectUri: 'http://localhost:5173',
```
[the remark this answers is not in the document]
Comment (cerebrl) [reply packages/davinci-client/README.md:19]: We can change this to add `/callback.html`.

File: packages/davinci-client/src/lib/davinci/node.slice.ts [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+      return collectors || [];
+    })
+    .addCase(updateNodeValues, (state: any[], action) => {
+      const collector = state.find((collector) => collector.id === action.payload.id);
+      if (!collector) {
+        console.error('Collector not found');
+        // TODO: Handle error
+      }
+      if (collector.input.length > 1 && !action.payload.index) {
+        console.error('Index not provided');
```
[the remark this answers is not in the document]
Comment (cerebrl) [reply packages/davinci-client/src/lib/davinci/node.slice.ts:75]: I'm hesitant to internally couple anything from the SDK that requires the use of `Config` as it won't align with the new architectural guidelines. I'm trying to just import more pure'ish functions into this DaVinci module.

File: packages/davinci-client/src/lib/style.css [hunk 4]
```diff
@@ -0,0 +1,114 @@
+:root {
```
[the remark this answers is not in the document]
Comment (cerebrl) [reply packages/davinci-client/src/lib/style.css:1]: The only thing that needs to be moved into this repo are the things under the `./davinci` directory. Anything that was outside of that was only used for prototyping and demo.

[the remark this answers is not in the document]
Comment (cerebrl) [reply packages/davinci-client/src/lib/style.css:1]: Just to be clear, the DaVinci module will have no UI/HTML/view related code in it at all. So, this style file and the `./components` directory should not be migrated over.

**Review by cerebrl — APPROVED** [review]

File: packages/javascript-sdk/package.json [hunk 5]
```diff
@@ -13,6 +13,12 @@
   "homepage": "https://github.com/ForgeRock/forgerock-javascript-sdk#readme",
   "main": "./src/index.ts",
   "module": "./src/index.ts",
-  "exports": "./src/index.ts",
+  "exports": {
+    ".": {
+      "import": "./src/index.ts",
+      "require": "./src/index.ts"
+    },
+    "./src/oauth2-client/state-pkce": "./src/oauth2-client/state-pkce"
```
Comment (ryanbas21) [inline packages/javascript-sdk/package.json:21]: had to add this for building. it isnt in the final output. but something to consider if we arent exporting this out of the root index

Comment (cerebrl) [reply packages/javascript-sdk/package.json:21]: Got it! I think keeping it mostly "internal" is actually good. I wouldn't want to publicly add more to the already large `index.ts` file.

**Pull request merged**